TalkAloud/TalkAloud/AudioEngine.swift

```diff
@@ -9,21 +9,52 @@
 import Foundation
 import AVFoundation
 
-// Struct will make it so I don't have to intialize variables
-// until I need to
-class AudioEngine {
-    var audioRecorder: AVAudioRecorder!
-    var audioPlayer: AVAudioPlayer!
-    var recordingSession: AVAudioSession!
-    var fileName = "selftalkfile.m4a"
+class AudioEngine: NSObject {
+    private var audioRecorder: AVAudioRecorder!
+    private var audioPlayer: AVAudioPlayer!
+    private var recordingSession: AVAudioSession!
+    private var fileName = "selftalkfile.m4a"
+    public private(set) var audioState: AudioEngineState = .stopped
+    private let audioRecordingSession = AVAudioSession.sharedInstance()
+    
+    public func getAudioRecorder() -> AVAudioRecorder {
```

Comment:
Remove these functions. Encapsulate all the logic for the AVAudioRecorder.

Reply:
Does this include the `getAudioRecordingSession`?